src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -493,8 +561,56 @@ private static List<List<ObjectIdentifier>> getContainedObjectBatches(S3AsyncCli
      */
     @Override
     public void copy(Path source, Path target, CopyOption... options) throws IOException {
+        //
+        // TODO: source and target can belong to any file system, we can not
```

Comment:
agreed, it was a old comment, I think I also read that copy between different sources may not be supported. I will try it out. For now I would keep the comment (maybe rephrasing it so that we have a TODO).